pkg/artifactory/resource_artifactory_remote_vcs_repository.go

```diff
@@ -0,0 +1,56 @@
+package artifactory
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+type VcsRemoteRepo struct {
+	RemoteRepositoryBaseParams
+	VcsGitProvider     string `json:"vcsGitProvider"`
+	MaxUniqueSnapshots int    `json:"maxUniqueSnapshots"`
+}
+
+func resourceArtifactoryRemoteVcsRepository() *schema.Resource {
+	const packageType = "vcs"
+
+	var vcsRemoteSchema = mergeSchema(baseRemoteRepoSchema, map[string]*schema.Schema{
+		"vcs_git_provider": {
+			Type:             schema.TypeString,
+			Optional:         true,
+			Default:          "GITHUB",
+			ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{"GITHUB", "BITBUCKET", "OLDSTASH", "STASH", "ARTIFACTORY", "CUSTOM"}, false)),
+			Description: "Artifactory supports proxying the following Git providers out-of-the-box: GitHub, Bitbucket, " +
+				"Stash, a remote Artifactory instance or a custom Git repository. Default value is 'GITHUB'.",
```

Comment:
Let's include the allowed values in the description. Same applies to the documentation file.

Reply:
Added